### PR TITLE
Lesson 1 8

### PR DESCRIPTION
**Pull request by Victoria240**
Data Fetching, Re-Fetching, and Memoized Handlers

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,63 +1,103 @@
...
-        <AddTodoForm onAddTodo={addTodo} />
-        {isLoading ? <p>Loading...</p> : <TodoList todoList={todoList} onRemoveTodo={removeTodo} />}
-        
 
-      </>
-    );
+    } catch (error) {
+      console.error('Error adding todo:', error.message);
+    }
+  }
```
Comment (dargaCode) [inline src/App.js:84]: This extra curly brace is one of the errors that's preventing the code from running: 

```suggestion
```

File: src/App.js [hunk 2]
```diff
@@ -1,63 +1,103 @@
...
+    <>
+      <h1>Todo List</h1>
+      {/* Pass the handleAddTodo function to the AddTodoForm component */}
+      <AddTodoForm onAddTodo={handleAddTodo} />
+      {/* Conditional rendering based on isLoading state */}
+      {isLoading ? (
+        <p>Loading...</p>
+      ) : (
+        // Pass the todoList and removeTodo function to the TodoList component
+        <TodoList todoList={todoList} onRemoveTodo={removeTodo} />
```
Comment (dargaCode) [inline src/App.js:97]: You've deleted the `removeTodo` function as part of this PR, which is the other error causing the page to crash

File: src/App.js [hunk 3]
```diff
@@ -1,63 +1,103 @@
...
-  }, []); // Empty dependency array, so this effect runs once after initial render
-
-
-    useEffect(() => {
-      if (!isLoading) {
-        localStorage.setItem('savedTodoList', JSON.stringify(todoList)); //save the todoList as a string in the localStorage whenever it changes 
+  // Define states for the list of todos and loading state
+  const [todoList, setTodoList] = useState([]); // Initialize todoList state with an empty array
+  const [isLoading, setIsLoading] = useState(true); // Initialize isLoading state as true
+  const [appState, setAppState] = useState ('')
```
Comment (dargaCode) [inline src/App.js:9]: I think I can see that you're trying to combine your two states into a single complex state object, but is that something the assignment is asking you to do? 

The code here is in an incomplete transition between these two concepts - You've got the 2 old states, and the new combined state, and I don't think the app could ever work unless it's fully one way or the other.

File: src/App.js [hunk 4]
```diff
@@ -1,63 +1,103 @@
...
-      const updatedTodoList = todoList.filter((todo) => todo.id !== id);
+      // Update the todoList state with fetched todos
+      setTodoList(todos);
+      setIsLoading(false); // Set isLoading to false after data fetch
 
-      // Update the todoList state with the new array of todos
-      setTodoList(updatedTodoList);
+    } catch (error) {
+      console.error('Error fetching data:', error.message);
     }
```
Comment (dargaCode) [inline src/App.js:45]: This function no longer existing is one of the things that's causing a page crash

**Review by dargaCode — CHANGES_REQUESTED** [review]
Hi Victoria, I know you said you're having some trouble on this assignment, and your code has a couple of crash bugs that are preventing it from running when I `npm start` it. 

Can you please take a look at these first couple of comments to remove the crashes? Then you should be able to test out whether the fetching/adding/deleting is working. 

Since there are a lot of moving pieces here, it's probably a good idea to book some time with me or one of the other mentors and dig into it together.

File: src/App.js (hunk 3 above)
Comment (Victoria240) [reply src/App.js:9]: Thank you for the detailed feedback! To be honest, I was exhausted when I worked on this task and wasn't able to get it to work right. I'll start over and read the assignment instructions carefully. Due to timezone differences, it's been a struggle to join the mentor sessions. Can I schedule a meeting with you at an earlier time to fix this, if necessary?

**Pull request closed without merging**